gobblin-kafka/src/main/java/gobblin/kafka/schemareg/LiKafkaSchemaRegistry.java

```diff
@@ -0,0 +1,270 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.kafka.schemareg;
+
+import java.io.IOException;
+import java.util.Properties;
+
+import org.apache.avro.Schema;
+import org.apache.commons.httpclient.Header;
+import org.apache.commons.httpclient.HttpClient;
+import org.apache.commons.httpclient.HttpException;
+import org.apache.commons.httpclient.HttpStatus;
+import org.apache.commons.httpclient.methods.GetMethod;
+import org.apache.commons.httpclient.methods.PostMethod;
+import org.apache.commons.pool2.impl.GenericObjectPool;
+import org.apache.commons.pool2.impl.GenericObjectPoolConfig;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Preconditions;
+
+import gobblin.configuration.ConfigurationKeys;
+import gobblin.kafka.serialize.MD5Digest;
+import gobblin.util.AvroUtils;
+
+
+/**
+ * Integration with LinkedIn's implementation of a schema registry that uses md5-hash for schema ids.
+ */
+public class LiKafkaSchemaRegistry implements KafkaSchemaRegistry<MD5Digest> {
```

Comment:
Essentially.. there will be a few more PR-s later to clean up the other kafka-related code, including KafkaAvroExtractor, I wanted to keep this PR relatively clean (without refactoring existing code) so that the bar for merging was low :)